pkg/gcptarget/gcptarget.go

```diff
@@ -0,0 +1,117 @@
+package gcptarget
+
+import (
+	"bytes"
+	"encoding/json"
+	"text/template"
+
+	"github.com/forseti-security/config-validator/pkg/api/validator"
+	asset2 "github.com/forseti-security/config-validator/pkg/asset"
+	"github.com/gogo/protobuf/jsonpb"
+	"github.com/open-policy-agent/frameworks/constraint/pkg/client"
+	"github.com/open-policy-agent/frameworks/constraint/pkg/types"
+	"github.com/pkg/errors"
+	"k8s.io/apiextensions-apiserver/pkg/apis/apiextensions"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+)
+
+// Name is the target name for GCPTarget
+const Name = "validation.gcp.forsetisecurity.org"
+
+// GCPTarget is the constraint framework target for FCV
+type GCPTarget struct {
+}
+
+var _ client.TargetHandler = &GCPTarget{}
+
+// New returns a new GCPTarget
+func New() *GCPTarget {
+	return &GCPTarget{}
+}
+
+// MatchSchema implements client.MatchSchemaProvider
+func (g *GCPTarget) MatchSchema() apiextensions.JSONSchemaProps {
+	return apiextensions.JSONSchemaProps{
+		Properties: map[string]apiextensions.JSONSchemaProps{
+			"target": {
+				Type: "array",
+				Items: &apiextensions.JSONSchemaPropsOrArray{
+					Schema: &apiextensions.JSONSchemaProps{
+						Type: "string",
+					},
+				},
+			},
+			"exclude": {
+				Type: "array",
+				Items: &apiextensions.JSONSchemaPropsOrArray{
+					Schema: &apiextensions.JSONSchemaProps{
+						Type: "string",
+					},
+				},
+			},
+		},
+	}
+}
+
+// GetName implements client.TargetHandler
+func (g *GCPTarget) GetName() string {
+	// Done
+	return Name
+}
+
+// Library implements client.TargetHandler
+func (g *GCPTarget) Library() *template.Template {
+	// TODO - lib impl
+	return libraryTemplate
+}
+
+// ProcessData implements client.TargetHandler
+func (g *GCPTarget) ProcessData(obj interface{}) (bool, string, interface{}, error) {
+	return false, "", nil, errors.Errorf("Storing data for referential constraint eval is not supported at this time.")
+}
+
+// HandleReview implements client.TargetHandler
+func (g *GCPTarget) HandleReview(obj interface{}) (bool, interface{}, error) {
+	switch asset := obj.(type) {
+	case *validator.Asset:
+		return g.handleAsset(asset)
+	case map[string]interface{}:
+		return true, asset, nil
```

Comment:
Is there something (field, value, etc.) we can check to make sure it's the correct map[string]interface{} (e.g., not just some random unmarshalled JSON)?

Reply:
done